upgrade_logic/business_objects/gateway.py

```diff
@@ -9,6 +9,45 @@
 from submodules.model import enums
 
 
+def gateway_1_10_1() -> bool:
+    __gateway_1_10_1_add_additional_embedding_information()
+    return True
+
+
+def __gateway_1_10_1_add_additional_embedding_information() -> bool:
+    def __transform_embedding_by_name(embedding_name: str):
+        splitted_name = embedding_name.split("-")
+        attribute_name = splitted_name[0]
+        embedding_type = splitted_name[1]
+        model = "-".join(splitted_name[2:])
+        if "bag-of-words" == model or "bag-of-characters" == model or "tf-idf" == model:
+            platform = enums.EmbeddingPlatform.PYTHON.value
+        else:
+            platform = enums.EmbeddingPlatform.HUGGINGFACE.value
+        name = f"{attribute_name}-{embedding_type}-{platform}-{model}"
+        return platform, model, name
+
+    print(
+        "Infer platform and new name and model out of old embedding name and add it to the embedding table ",
+        flush=True,
+    )
+    embeddings = embedding.get_all_embeddings()
+    idx = 0
+    for embedding_item in embeddings:
+        if embedding_item.platform:
+            continue
+        idx += 1
+        platform, model, name = __transform_embedding_by_name(embedding_item.name)
+        embedding_item.platform = platform
+        embedding_item.model = model
+        embedding_item.name = name
+
+    general.commit()
+    print(f"Added new embedding information for {idx} records", flush=True)
```

Comment:
to prevent irritation i think i'd go with Updated information for xxx embeddings (since I first though records meant embedding tensor entries)

Reply:
changed